### PR TITLE
add serialization methods and cn-client codes for pipeline's remote-run

### PR DESCRIPTION
**Pull request by m-schen**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

related to issue #3591

## What this PR does / why we need it:
1. add serialization methods for struct Compile.Scope for pipeline's remote run.
2. add cn-client and cn-service related methods.

### CODE REVIEW

**Review by daviszhen — APPROVED** [review]

**Review by reusee — APPROVED** [review]

File: pkg/cnservice/server.go [hunk 1]
```diff
@@ -87,15 +94,16 @@ func (s *service) Close() error {
...
+
+func defaultRequestHandler(ctx context.Context, message morpc.Message, cs morpc.ClientSession) error {
+	return nil
 }
 
-/*
-func (s *service) releaseMessage(msg *pipeline.Message) {
-	msg.Reset()
-	s.pool.Put(msg)
+func (s *service) handleRequest(ctx context.Context, req morpc.Message, _ uint64, cs morpc.ClientSession) error {
```
Comment (zhangxu19830126) [inline pkg/cnservice/server.go:104]: SCA Error, unused

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

File: pkg/cnservice/server.go (hunk 1 above)
Comment (m-schen) [reply pkg/cnservice/server.go]: it's really unused now. I will comment it.

**Comment by m-schen** [comment]
@matrix-meow rerun failed

**Comment by m-schen** [comment]
@matrix-meow rerun failed

**Comment by m-schen** [comment]
@matrix-meow rerun failed

**Review by zhangxu19830126 — APPROVED** [review]

**Pull request merged**